### ISSUE TITLE
How can I check that if basic requirements are met before installing my plugin?

### ISSUE THREAD

**Issue by lonalore** [e107inc/e107#879]
How can I check that if basic requirements are met before installing my plugin. And how can I break plugin installation process if something is missing (e.g. my plugin depends on another plugin, but it's not present, or it's not installed)?

I think, I should use install_pre() method in PLUGIN_setup() class in the PLUGIN_setup.php to check plugin requirements, but I dont know how I can cancel the installation process if something is missing. Can I break it easily?

Maybe, I should define plugin dependencies in the plugin.xml?

Thanks :)

**Comment by CaMer0n** [comment]
@lonalore can you give me an example of requirements? There are many automated checks in the plugin installation process. Are you referring more to upgrades rather than fresh installations of your plugin?

**Comment by lonalore** [comment]
I am referring fresh installations. Well, I am working on a Node.js integration plugin, which provides an API that other plugins can use to add realtime capabilities to e107, specifically enabling pushing updates to open connected clients. After it has a release, I am going to start creating (sub-) plugins, and I would like to add the main "nodejs" plugin as a dependency to them.

**Comment by CaMer0n** [comment]
I have added an advanced example to the docs. See 'dependencies' in plugin.xml
http://e107.org/developer-manual/plugin-basics#setup-and-installation

**Comment by lonalore** [comment]
Thank you! It helped me a lot! :)

**Comment by lonalore** [comment]
Maybe I found a bug in plugin-installation process:

"... There are many automated checks in the plugin installation process. ..." 

Which file contains these kind of checks? Because I defined some (plugin) dependencies in my plugin.xml, e.g: https://github.com/lonalore/nodejs_pm/blob/master/plugin.xml
... but e107 allows me to install the plugin even if required plugins are missing/not installed. 

I think e107 should throw an error if something is missing, or simply disable the checkbox in the current row.
